### PR TITLE
vscodium: add uninstall, depends_on, update zap

### PR DESCRIPTION
**Pull request by toobuntu**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by toobuntu** [comment]
The build identifier changed in 1.71.0 (Sep 2, 2022) to `com.vscodium` instead of `com.visualstudio.code.oss`. The current release is 1.85.1 (Dec 13, 2023).

I updated the zap accordingly. But I'm not sure if the old entries should remain for a while in case users still have a really old version installed.

From the [1.71.0 release notes](https://github.com/VSCodium/vscodium/releases/tag/1.71.0.22245):
> - Warning: Due to internal changes of build identifiers a manual update on MacOS is needed; on Win32 you may want to uninstall the old and install the new version - see https://github.com/VSCodium/vscodium/issues/1227

File: Casks/v/vscodium.rb [hunk 1]
```diff
@@ -11,17 +11,21 @@
...
+
   zap trash: [
     "~/.vscode-oss",
-    "~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/com.visualstudio.code.oss.sfl*",
     "~/Library/Application Support/VSCodium",
-    "~/Library/Logs/VSCodium",
-    "~/Library/Preferences/com.visualstudio.code.oss.helper.plist",
-    "~/Library/Preferences/com.visualstudio.code.oss.plist",
-    "~/Library/Saved Application State/com.visualstudio.code.oss.savedState",
+    "~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/com.vscodium.sfl3",
```
Comment (krehel) [inline Casks/v/vscodium.rb:24]:
```suggestion
    "~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/com.vscodium.sfl*",
```

Comment (krehel) [reply Casks/v/vscodium.rb:24]: We typically wildcard these

Comment (toobuntu) [reply Casks/v/vscodium.rb]: Done!

**Comment by toobuntu** [comment]
I didn't have a `~/Library/Preferences/com.vscodium.helper.plist`, but its counterpart was listed (with the old bundle ID). So I wildcarded that, too.

**Comment by krehel** [comment]
That works.  Thanks @toobuntu!

**Review by krehel — APPROVED** [review]

**Pull request merged**